lib/bolt/transport/local/shell.rb

```diff
@@ -20,17 +20,19 @@ def initialize(target)
           @user = ENV['USER'] || Etc.getlogin
           @run_as = target.options['run-as']
           @logger = Logging.logger[self]
+          @sudo_id = SecureRandom.uuid
+          @send_stdin = true
```

Comment:
I was not able to trigger data to be sent twice, but I left this in to guard against it. Can we safely remove it?

Reply:
I think this can only happen if `sudo randomchars` is in stdout twice, which would only happen if that *happened* to be output by a script that was running. That could happen before or after the actual sudo check though, so I don't think we gain anything from checking if it's been run already - if `sudo randomchars` is on stdout *not* from us, behavior is unexpected.

Reply:
Also once stdin is sent we close that channel. Just to clarify we send and look for the `sudo_id` on stderr.